src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        (uint256 pid, address rewardsPool, ERC20 lpt, CurvePool pool, bytes4 selector) = abi.decode(
+            adaptorData,
+            (uint256, address, ERC20, CurvePool, bytes4)
+        );
+
+        if (selector != bytes4(0)) _callReentrancyFunction(pool, selector);
+        else revert BaseAdaptor__UserDepositsNotAllowed();
+
+        _validatePositionIsUsed(pid, rewardsPool, lpt, pool, selector);
+        lpt.safeApprove(address(booster), assets);
+
+        booster.deposit(pid, assets, true);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(lpt, address(booster));
+    }
+
+    /**
+     * @notice If a user withdraw needs more LPTs than what is in the Cellar's wallet, then the Cellar will unstake, unwrap cvxLPTs, and withdraw LPTs from Convex
+     * @param amount of LPT to unstake, unwrap, and withdraw from Convex market
+     * @param receiver see baseAdaptor.sol
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdraw(
+        uint256 amount,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public override {
+        // Check that position is setup to be liquid.
+        bool isLiquid = abi.decode(configurationData, (bool));
+
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        (uint256 pid, address rewardPool, ERC20 lpt, CurvePool pool, bytes4 selector) = abi.decode(
+            adaptorData,
+            (uint256, address, ERC20, CurvePool, bytes4)
+        );
+
+        if (isLiquid && selector != bytes4(0)) _callReentrancyFunction(pool, selector);
+        else revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        _validatePositionIsUsed(pid, rewardPool, lpt, pool, selector);
+
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(rewardPool);
+        ERC20 stakingToken = ERC20(baseRewardPool.stakingToken());
+
+        if (amount <= stakingToken.balanceOf(msg.sender)) {
+            stakingToken.safeTransfer(receiver, amount);
+        } else {
+            baseRewardPool.withdrawAndUnwrap(amount, false);
+        }
+    }
+
+    /**
+     * @notice Functions Cellars use to determine the withdrawable balance from an adaptor position.
+     * @dev Accounts for LPTs staked in Convex Market from calling Cellar.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
```

Comment:
Change to `if (isLiquid && selector != bytes4(0)) {`

Reply:
Done

Reply:
Done referencing `CurveAdaptor.sol` code [here](https://github.com/PeggyJV/cellar-contracts/blob/76b839a2b9e37ee551b8cc0698cdfbada6cd66df/src/modules/adaptors/Curve/CurveHelper.sol#L317).